internal/pkg/apply/apply.go

```diff
@@ -39,13 +42,77 @@ type Result struct {
 // Do executes the apply
 func (a *Apply) Do() (Result, error) {
 	fmt.Fprintf(a.Out, "Doing `cli-experimental apply`\n")
-	pods, err := a.Clientset.CoreV1().Pods("default").List(metav1.ListOptions{})
-	if err != nil {
-		return Result{}, err
-	}
-	for _, p := range pods.Items {
-		fmt.Fprintf(a.Out, "Pod %s\n", p.Name)
-	}
 
+	// TODO(Liuijngfang1): add a dry-run for all objects
+	// When the dry-run passes, proceed to the actual apply
+
+	for _, u := range a.Resources {
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(u.GroupVersionKind())
+		err := a.Client.Get(context.Background(), client.ObjectKey{
+			Namespace: u.GetNamespace(),
+			Name:      u.GetName(),
+		}, obj)
+
+		if err != nil && errors.IsNotFound(err) {
+			err = a.Client.Create(context.Background(), &u)
```

Comment:
Lets not do this.  We need to use server side apply or the existing client-side apply code.